rust/lance-index/src/vector/pq/distance.rs

```diff
@@ -80,9 +80,11 @@ pub(super) fn compute_l2_distance(
     // so code[i * num_vectors + j] is the code of i-th sub-vector of the j-th vector.
     let num_vectors = code.len() / num_sub_vectors;
     let mut distances = vec![0.0_f32; num_vectors];
-    let num_centroids = 2_usize.pow(num_bits);
+    // it must be 8
+    const NUM_CENTROIDS: usize = 2_usize.pow(8);
```

Comment:
nit: seems like this would be cleaner if we refactor `compute_l2_distance` to a `trait ProductQuantizedL2Distance<const BIT_WIDTH: usize>`. We can probably avoid branching in pq dist calculator this way too. Just a ticket for now is fine

Reply:
yeah i was thinking of this way as well, but 4bit PQ impl is diff from 8bit, so still 2 methods